### PR TITLE
Category & Breadcrumb filters, use Breadcrumb module, Fix Breadcrumb Homepage link

### PR DESCRIPTION
**Pull request by grimasod**
### Short description and why it's useful

Pre-define category search query filters in local.json that are applied to every query. This removes the need to define more and more parameters for category/list calls, but still allows parameters to override the pre-defined filters. It also means the currently hardcoded platform-specific parameters, like "is_active" and "product_count", could be deprecated.

### Which environment this relates to
Check your case. In case of any doubts please read about [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html)

- [x] Test version (https://test.storefrontcloud.io) - this is a new feature or improvement for Vue Storefront. I've created branch from `develop` branch and want to merge it back to `develop`
- [ ] RC version (https://next.storefrontcloud.io) - this is a stabilisation fix for Release Candidate of Vue Storefront. I've created branch from `release` branch and want to merge it back to `release`
- [ ] Stable version (https://demo.storefrontcloud.io) - this is an important fix for current stable version. I've created branch from `hotfix` or `master` branch and want to merge it back to `hotfix`

### Upgrade Notes and Changelog

- [x] No upgrade steps required (100% backward compatibility and no breaking changes)
- [ ] I've updated the [Upgrade notes](https://github.com/DivanteLtd/vue-storefront/blob/develop/doc/Upgrade%20notes.md) and [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md) on how to port existing VS sites with this new feature

**IMPORTANT NOTICE** - Remember to update `CHANGELOG.md` with description of your change

### Contribution and currently important rules acceptance

- [x] I read and followed [contribution rules](https://github.com/DivanteLtd/vue-storefront/blob/master/CONTRIBUTING.md)
- [x] I read the [TypeScript Action Plan](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/TypeScript%20Action%20Plan.md) and adjusted my PR according to it
- [x] I read about [Vue Storefront Modules](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/api-modules/about-modules.md) and I am aware that every new feature should be a module

### CODE REVIEW

**Comment by grimasod** [comment]
(Remake of previous PR, now on correct branch)

I'm not certain that this is the correct approach, so this is WIP. I'll wait for confirmation before updating the Changelog, etc.

**Comment by grimasod** [comment]
Sample portion of local.json
```
"entities": {
    "optimize": true,
    "twoStageCaching": true,
    "optimizeShoppingCart": true,
    "category": {
      "includeFields": [ ... ],
      "excludeFields": [ ... ],
      "filterFields": [
        {
          "key": "include_in_menu",
          "value": { "eq": true }
        },
        {
          "key": "is_active",
          "value": { "eq": true }
        },
        {
          "key": "product_count",
          "value": { "gt": 0 }
        }
      ],
      ...
    },
},
```

**Comment by pkarw** [comment]
This is for sure good idea @grimasod. Please just note that from 1.11 (`develop` branch) there is a `catalog-next` module which is in charge of displaying Category pages. The changes you've applied are to `catalog`. Is it intended? I think it should be applied to `catalog-next`. @patzick

**Comment by grimasod** [comment]
@pkarw Thanks for pointing that out. I'm already using this patch on our live site (v1.9) and just copied it to the develop branch. I'll test with `catalog-next` and then update here

File: core/data-resolver/CategoryService.ts [hunk 1]
```diff
@@ -16,6 +16,7 @@ const getCategories = async ({
   includeFields = config.entities.optimize ? config.entities.category.includeFields : null,
   excludeFields = config.entities.optimize ? config.entities.category.excludeFields : null
 }: DataResolver.CategorySearchOptions = {}): Promise<Category[]> => {
+  const allFilters = 'filterFields' in config.entities.category ? Object.assign(config.entities.category.filterFields, filters) : filters
```
Comment (patzick) [inline core/data-resolver/CategoryService.ts]:
```suggestion
  const allFilters = config.entities.category.filterFields ? Object.assign({}, config.entities.category.filterFields, filters) : filters
```
we shouldn't try to assign into config property

File: core/modules/catalog/store/category/actions.ts [hunk 2]
```diff
@@ -39,6 +39,14 @@ const actions: ActionTree<CategoryState, RootState> = {
     const commit = context.commit
     let customizedQuery = false // that means the parameteres are != defaults; with defaults parameter the data could be get from window.__INITIAL_STATE__ - this is optimisation trick
     let searchQuery = new SearchQuery()
+
+    if ('filterFields' in rootStore.state.config.entities.category) {
```
Comment (patzick) [inline core/modules/catalog/store/category/actions.ts]:
```suggestion
    if (config.entities.category. filterFields) {
```

File: core/modules/catalog/store/category/actions.ts [hunk 3]
```diff
@@ -39,6 +39,14 @@ const actions: ActionTree<CategoryState, RootState> = {
...
+      for (const field of rootStore.state.config.entities.category.filterFields) {
```
Comment (patzick) [inline core/modules/catalog/store/category/actions.ts]:
```suggestion
      for (const field of config.entities.category.filterFields) {
```

**Comment by grimasod** [comment]
@patzick I'm in the middle of converting our site to v1.10, so haven't had time to work on this. This PR should be totally compatible with the old module, I'll test it soon. But it's on the develop branch, so I need to convert it to work on `catalog-next`. Is it correct that we should only use `catalog-next` from v.1.11? Or do we need to update both?

File: core/modules/catalog/store/category/actions.ts (hunk 2 above)
Comment (grimasod) [reply core/modules/catalog/store/category/actions.ts]: I need to redo this, it has a conflict

File: core/modules/catalog/store/category/actions.ts (hunk 3 above)
Comment (grimasod) [reply core/modules/catalog/store/category/actions.ts]: I need to redo this, it has a conflict

**Comment by grimasod** [comment]
@patzick @pkarw Hi guys, the develop branch, the sidebar menu still uses the `category` store, not the new `category-next` store. Is this intentional or WIP? Thanks

**Comment by patzick** [comment]
Hey @grimasod, no, it should use just category-next, but main focus was on the category page itself :) I will change that eventually, or you can if you want to with this PR :)

**Comment by patzick** [comment]
hey @grimasod, what do we have here? :) are we proceeding further? We have long inactivity here

**Comment by grimasod** [comment]
Hey @patzick The router manager took up a lot of time and I've been catching up on other work. Just got back to this today.
The changes to category search in catalog-next are really great, but this update isn't compatible in it's current form. I'm looking at how it works now, to see if I can apply this.

I'm also looking at a "breadcrumbFilterFields" config setting, which would only apply to filtering categories for breadcrumbs

**Comment by grimasod** [comment]
@patzick @pkarw This PR evolved to work with catalog-next and use the breadcrumb module. I know it's a bit late, but I'm hoping this can be included in 1.11.0-RC1.

Currently on the Develop branch there are some issues with both catalog and catalog-next being used simultaneously. In particular, breadcrumb categories when products are in multiple categories - they can end up different in catalog and catalog-next, resulting in no breadcrumbs displaying.

This PR addresses these issues too.

**Comment by grimasod** [comment]
@patzick I need to close this PR and create a new one on the release/v1.11 branch instead

**Comment by grimasod** [comment]
See #3691

**Pull request closed without merging**